src/routes/data/handlers.test.ts

```diff
@@ -163,8 +168,28 @@ describe('Data routes', () => {
         });
     });
 
-    // Multiple ranges are not yet supported
-    it('should return 416 status code for a range request with multiple ranges', async () => {
+    it('should return 416 status code for a unsatisfiable range request', async () => {
```

Comment:
Perhaps this should be two separate tests:
- one is an unsatisfiable range because the tx doesn't exist
- the other is an unsatisfiable range because it's beyond the bounds of the tx size (this could actually be 2 tests - one where just the end of the range is invalid and one where the start AND the end of the range are invalid)

Reply:
When tx doesn't exist, we respond with a 404. The check happens before `handleRangeRequest` get into action.